### ISSUE TITLE
Documentation for `check_output_custom`

### ISSUE THREAD

**Issue by elijahbenizzy** [DAGWorks-Inc/hamilton#212]
**Is your feature request related to a problem? Please describe.**
Custom output-checking is something people want, but we don't make it easy. Closes is unit tests: https://github.com/DAGWorks-Inc/hamilton/blob/e55cb4daed0aa3fc38fe338d606470ebe1371127/tests/function_modifiers/test_validation.py#L132.  Otherwise it says "ask in slack"

**Describe the solution you'd like**
Documentation in `readthedocs`

**Additional context**
https://hamilton-opensource.slack.com/archives/C03M33QB4M8/p1688457568511989

**Comment by skrawcz** [comment]
Task - add section to docs on data quality using this.

**Comment by audacious-fk** [comment]
Hi! Is this issue open?

**Comment by skrawcz** [comment]
@audacious-fk yes it is. You might find this [blog helpful](https://blog.dagworks.io/p/data-quality-with-hamilton-and-pandera?utm_source=publication-search). 
Do you know what your task should be here? 
For me it would be to [add to this section](https://hamilton.dagworks.io/en/latest/how-tos/run-data-quality-checks/).

**Comment by audacious-fk** [comment]
Thanks so much! I’m new to open source, so I really appreciate the guidance!
I'll start working on this.
